Decimal number separator fix

pixelshade opened the pull request:
Set globally in constructor

pixelshade committed:
Decimal number separator fix

Set globally in constructor

ms7s commented on source/Assets/Scripts/InfinarioSDK/Infinario.cs:
```diff
@@ -15,6 +15,11 @@ private Infinario()
 			implementation = new SDK.Unity ();
+
+			//Setup decimal separator for numbers
+            var customCulture = (System.Globalization.CultureInfo)System.Threading.Thread.CurrentThread.CurrentCulture.Clone();
```
Indentation here (spaces) does not match the surrounding code (tabs). Please use tabs so the formatting is consistent.

pixelshade committed:
Tabs as indentation, better comment.

Using tabs instead of spaces. Added more info about issue fix in the
comment.

ms7s commented on source/Assets/Scripts/InfinarioSDK/Infinario.cs:
```diff
@@ -15,6 +15,11 @@ private Infinario()
+			//Setup decimal separator for numbers
+            var customCulture = (System.Globalization.CultureInfo)System.Threading.Thread.CurrentThread.CurrentCulture.Clone();
+            customCulture.NumberFormat.NumberDecimalSeparator = ".";
+            System.Threading.Thread.CurrentThread.CurrentCulture = customCulture;    
```
You can't change a thread's culture from within the SDK, there are two reasons for that:
- You may affect the host application, which may have set their own culture
- The SDK runs in multiple threads and you are changing the culture for only one of them

Please use the correct Culture explicitly at the place where the string gets formatted. You want to use `CultureInfo.InvariantCulture` as the culture as it already has `.` as  the decimal separator. I'm not sure about thousands separator though, please check that.

Commit:
Invariant culture set in double converting to string

Invariant culture set in double converting to string in miniJSON due to
system locale bug where “,” decimal separator was used instead of “.”

Commit:
Culture info set invariant, separator set to dot

Its questionable if its ok to call it in method everytime

Commit:
Decimal conversion using Invariant Culture with R param

Decimal conversion using Invariant Culture with R parameter that should
be enough to fix problem with “,” in decimals in some cultures
